lib/auth/jwtclient.ts

```diff
@@ -0,0 +1,242 @@
+/**
+ * Copyright 2014 Google Inc. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import Auth2Client from './oauth2client';
+import * as gToken from 'gtoken';
+import JWTAccess from './jwtaccess';
+import { isString } from 'lodash';
+import * as util from 'util';
+
+const noop = Function.prototype;
+
+export default class JWT extends Auth2Client {
+
+  public email: string;
+  public keyFile: string;
+  public key: string;
+  public scopes: string | string[];
+  public subject: string;
+  public gToken: any;
```

Comment:
I'm not sure that I understand the question.  I kept the public properties here because they were on the existing API contract, and I'm trying to figure out how to not break it if possible :(